### PR TITLE
feat: GovWallet custom theme

### PR DESCRIPTION
**Pull request by dalsontws**
[Notion link](https://www.notion.so/sally-wallet/UI-Label-change-Leverage-Sally-app-to-capture-payment-QR-for-merchant-ringfencing-901854c276cc4c4991d8e1765e63e1e5) 

This PR adds... 
- GovWallet custom theme

The GovWallet theme consists of a Red-white colour scheme, configured using the `ThemeContextProvider` and dependent on the new feature flag `theme`
- Defaults to SallyTheme (no changes apart from background, noted below)
- Upon fetching of the campaign config, the theme will be set based on the feature flag
- At each related component, the `useTheme` hook will provide the current theme 

Notable changes to the existing theme 
- background colour of app to use [#F5F5F5](https://www.color-hex.com/color/f5f5f5) instead of default white ([#FFFFFF](https://www.color-hex.com/color/ffffff)) to provide more contrast between the Cards(e.g. CustomerCard, LoginScanCard etc) and the background

- [x] I've kept this PR as small as possible (~600 lines) by splitting it into PRs with manageable chunks of code
- [ ] I've requested reviews from 1-2 reviewers
- [ ] I've added [accessibility IDs](https://www.notion.so/dc5f4ce910f7431c84344ac79344e9f5?v=d487366816834dd4ab8dc12e0b5928f3) to my components
- [ ] I've added/updated unit/integration tests
- [ ] I've added/updated [e2e tests](https://www.notion.so/e2e-Documentation-a096d3e0bf75485e85ad692af8371ef3) for at least the happy flow
- [ ] I've added [Chinese translations for i18n setup](https://www.notion.so/Translations-Dev-Guide-8c1da838982a4f59a386ec96ac6780c8)
- [ ] I've added documentation in README/[Notion](https://www.notion.so/82b92fb1007640328dab9582c0a3694e?v=3b6ce48202cf40ad8450553799b13146)

### CODE REVIEW

**Review by sebastianquek — COMMENTED** [review]
Looks good! Just one minor comment.

In addition, is this a breaking change?

**Comment by dalsontws** [comment]
> In addition, is this a breaking change?

doubt so since the feature flag `theme` is optional, and the app theme itself defaults to the existing colours!

File: src/types.ts [hunk 1]
```diff
@@ -183,6 +183,7 @@ const CampaignFeatures = t.intersection([
   t.partial({
     alternateIds: t.array(IdentificationFlag),
     apiVersion: t.union([t.literal("v1"), t.literal("v2")]),
+    theme: t.union([t.literal("DEFAULT"), t.literal("GOV_WALLET")]),
```
[the remark this answers is not in the document]
Comment (dalsontws) [reply src/types.ts]: made the change [here](https://github.com/rationally-app/mobile-application/pull/493/commits/8340f36f248904fd098036efcdbb039ab43ce0f5) as well as changes to the param store!

File: src/context/theme.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        setThemeValue(govWalletTheme);
+      } else {
+        setThemeValue(sallyTheme);
+      }
+    },
+    [setThemeValue]
+  );
+
+  return (
+    <ThemeContext.Provider value={{ theme: theme, setTheme }}>
```
Comment (sebastianquek) [inline src/context/theme.tsx:39]: Just noticed, can we do the following? Or does eslint complain?
```suggestion
    <ThemeContext.Provider value={{ theme, setTheme }}>
```

Comment (dalsontws) [reply src/context/theme.tsx]: oh yes we can! changed [here](https://github.com/rationally-app/mobile-application/pull/493/commits/ff9c4c50fd7bd9304fc5a2e589e259806460ef2b)!

File: src/common/styles/themes.tsx [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    disabledChevronOpacity: number;
+  };
+  alertModal: {
+    titleColor: string;
+  };
+};
+
+export const govWalletTheme: CustomTheme = {
+  name: "GOVWALLET",
+  // eslint-disable-next-line react/display-name
```
Comment (yongjiajie) [inline src/common/styles/themes.tsx:90]: Why is this necessary?

File: src/context/theme.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  theme: sallyTheme,
+  setTheme: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
+});
+
+export const ThemeContextProvider: FunctionComponent = ({ children }) => {
+  const [theme, setThemeValue] = useState(sallyTheme);
+
+  const setTheme = useCallback(
+    (themeName?: string) => {
+      if (themeName === "GOVWALLET") {
```
Comment (yongjiajie) [inline src/context/theme.tsx:29]: Just a minor nit: Would it be better if `DEFAULT` and `GOVWALLET` were global constants?

File: src/context/theme.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+} from "../common/styles/themes";
+
+interface ThemeContext {
+  theme: CustomTheme;
+  setTheme: (themeName: string | undefined) => void;
+}
+
+export const ThemeContext = createContext<ThemeContext>({
+  theme: sallyTheme,
+  setTheme: () => {}, // eslint-disable-line @typescript-eslint/no-empty-function
```
Comment (yongjiajie) [inline src/context/theme.tsx:21]: I saw a few other examples of this being:

```
setTheme: () => undefined

setTheme: () => void
```

Is there a difference between having an empty function and the above examples?

**Review by yongjiajie — COMMENTED** [review]
Great job! Just had some questions!

File: src/common/styles/themes.tsx (hunk 3 above)
Comment (dalsontws) [reply src/common/styles/themes.tsx:88]: the linter is complaining since the function returning the logo component isn't named. Wasn't sure if we needed to name this function tho since it's returning a static image! Wdyt?

File: src/context/theme.tsx (hunk 4 above)
Comment (dalsontws) [reply src/context/theme.tsx]: yup! changed [here](https://github.com/rationally-app/mobile-application/pull/493/commits/c4c55ee6d802eaf0e74c62b715427f7a90f889ce)!

File: src/context/theme.tsx (hunk 5 above)
Comment (dalsontws) [reply src/context/theme.tsx:22]: Hmm dont think there's a difference, good observation on this! Perhaps we should standardise the entire code in a subsequent PR?

File: src/common/styles/themes.tsx (hunk 3 above)
Comment (yongjiajie) [reply src/common/styles/themes.tsx:88]: I think it's fine!

**Review by sebastianquek — APPROVED** [review]
LGTM!! great job!!

**Review by yongjiajie — APPROVED** [review]
LGTM
